### PR TITLE
feat: add lcm of n numbers

### PR DESCRIPTION
**Pull request by Soptq**
This PR adds LCM of N numbers.

## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

- solving LCM of N numbers.
- all tests are passed.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/math/src/lcm_of_n_numbers.cairo [hunk 1]
```diff
@@ -0,0 +1,22 @@
+//! # LCM for N numbers
+
+// Calculate the lowest common multiple for n numbers
+// # Arguments
+// * `n` - The array of numbers to calculate the lcm for
+// # Returns
+// * `u128` - The lcm of input numbers
+use alexandria_math::gcd_of_n_numbers::gcd_two_numbers;
+
```
Comment (LucasLvy) [inline src/math/src/lcm_of_n_numbers.cairo:9]:
```suggestion
use alexandria_math::gcd_of_n_numbers::gcd_two_numbers;

/// Calculate the lowest common multiple for n numbers
/// # Arguments
/// * `n` - The array of numbers to calculate the lcm for
/// # Returns
/// * `u128` - The lcm of input numbers
```

File: src/math/src/lcm_of_n_numbers.cairo [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+fn lcm(mut n: Span<u128>) -> u128 {
```
Comment (LucasLvy) [inline src/math/src/lcm_of_n_numbers.cairo:10]: return a result don't panic

File: src/math/src/lcm_of_n_numbers.cairo [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    // Return empty input error
+    if n.is_empty() {
+        panic_with_felt252('EI')
```
Comment (LucasLvy) [inline src/math/src/lcm_of_n_numbers.cairo:13]:
```suggestion
        return Result::Err(....)
```

**Review by LucasLvy — CHANGES_REQUESTED** [review]
rename the tests to correct names, might also be worth to implement either lcm for any type that implements `Mul` and `Div`(this would require a change in the gcd function) or implement that function for any type that implements `TryInto<..., u128>`

File: src/math/src/lcm_of_n_numbers.cairo (hunk 1 above)
Comment (Soptq) [reply src/math/src/lcm_of_n_numbers.cairo]: Fixed ;)

File: src/math/src/lcm_of_n_numbers.cairo (hunk 3 above)
Comment (Soptq) [reply src/math/src/lcm_of_n_numbers.cairo]: Fixed ;)

File: src/math/src/lcm_of_n_numbers.cairo (hunk 2 above)
Comment (Soptq) [reply src/math/src/lcm_of_n_numbers.cairo]: Fixed ;)

**Comment by Soptq** [comment]
> rename the tests to correct names, might also be worth to implement either lcm for any type that implements `Mul` and `Div`(this would require a change in the gcd function) or implement that function for any type that implements `TryInto<..., u128>`

Thanks for the advice, it is now done ;)

**Review by LucasLvy — CHANGES_REQUESTED** [review]
Sorry i'm dumb i meant `Into<T, u128>` not `TryInto` can you change that plz

**Review by LucasLvy — APPROVED** [review]

**Pull request merged**